### ISSUE TITLE
Custom Field Type 'URL' does not refer to Target Value

### ISSUE THREAD

**Issue by rytechsites** [joomla/joomla-cms#43616]
### Steps to reproduce the issue
Create a custom field - URL
Allow for RELATIVE LINKS

### Expected result
If you put in an INTERNAL URL as the URL (relative link) it should open in the Current Browser, and not in a new browser.

### Actual result
On the click of the URL Field it always opens in a NEW TAB (even if it is an internal link)

### System information (as much as possible)
Joomla 4.4.5
PHP 8

### Additional comments

Here is the actual code:
if (!Uri::isInternal($value)) {
    $attributes = ' rel="nofollow noopener noreferrer" target="_blank"';
    $text       = Text::_('JVISIT_WEBSITE');
} else {
    $text       = Text::_('JVISIT_LINK');
}

It seems that in the code it is 'hardcoded' to always use _blank
1. although it 'asks' if URL is internal, even if you use a referential link (./sss/jkdf) when this code is run the URL is already 'the full url', so the system does not know that it is internal

**Comment by brianteeman** [comment]
Ignoring the fact that target="_blank" should never be forced

I can confirm that if the field is set to use relative urls then the isInternal check will never pass so the target will be added.

If you do NOT use relative urls then the isInternal check does work

**Comment by rytechsites** [comment]
Yes IsInternal does work if it is external.  But if it is an internal link then it does not work.  Is that what you are saying Brian?

Also I don't see the TARGET variable that you can define in the custom field build being referenced.  Should it be?  Similar to how you can select the target in a menu item?

thanks,
Laura<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/43616">issues.joomla.org/tracker/joomla-cms/43616</a>.</sub>

**Comment by brianteeman** [comment]
> Yes IsInternal does work if it is external. But if it is an internal link then it does not work. Is that what you are saying Brian?

Not quite

> Also I don't see the TARGET variable that you can define in the custom field build being referenced. Should it be? Similar to how you can select the target in a menu item?

**Comment by rytechsites** [comment]
Ok, got it.  Interesting not sure why I would 'not use relative' when it is an internal link... since that would be the purpose of using relative URLs, so I can use an internal link within the site.

Do you know what the variable is for 'target' that is defined in the form?  I could create a temporary override just for now until we have the fix.

-- Laura<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/43616">issues.joomla.org/tracker/joomla-cms/43616</a>.</sub>

**Comment by rytechsites** [comment]
It seems that the 'TARGET' is not part of core joomla, but there is still a bug...

I'm going to edit the request.
